lib/platform/github/index.ts

```diff
@@ -1735,3 +1744,79 @@ export async function getVulnerabilityAlerts(): Promise<VulnerabilityAlert[]> {
   }
   return alerts;
 }
+
+async function pushFiles({
+  branchName,
+  files,
+  message,
+}: CommitFilesConfig): Promise<CommitSha | null> {
+  const additions: FileAddition[] = [];
+  const deletions: FileDeletion[] = [];
+
+  for (const file of files) {
+    if (file.type === 'addition') {
+      additions.push(file);
+    } else {
+      deletions.push(file);
+    }
+  }
+
+  const executableFiles = additions.filter(({ isExecutable }) => isExecutable);
+  if (executableFiles.length) {
+    logger.warn(
+      { branchName, executableFiles: executableFiles.map(({ path }) => path) },
+      'Platform-native commit: found executable files'
+    );
+    return null;
+  }
+
+  const fileChanges = {
+    additions: additions.map(({ path, contents }) => ({
+      path,
+      contents: Buffer.from(contents).toString('base64'),
+    })),
+    deletions: deletions.map(({ path }) => ({ path })),
+  };
+
+  const variables = {
+    repo: config.repository,
+    repositoryId: config.repositoryId,
+    branchName: `refs/heads/${branchName}`,
+    oid: config.defaultBranchOid,
```

Comment:
What is the significance of this field? And what if our PR is for a different base branch than the default?

Reply:
Changed to `oldSha` obtained at the moment of local branch creation